config/file/parser.go

```diff
@@ -33,3 +33,22 @@ func newParser(ext extension) (configParser, error) {
 		return configParser{}, errors.New("unsupported config format")
 	}
 }
+
+// unmarshaledConfig is a raw data model for runner config files.
+// It serves as a receiver for unmarshaling processes and for that reason
+// its types are kept simple (certain types are incompatible with certain
+// unmarshalers).
+type unmarshaledConfig struct {
+	Request struct {
+		Method      *string           `yaml:"method" json:"method"`
+		URL         *string           `yaml:"url" json:"url"`
+		QueryParams map[string]string `yaml:"queryParams" json:"queryParams"`
+		Timeout     *string           `yaml:"timeout" json:"timeout"`
+	} `yaml:"request" json:"request"`
+
+	RunnerOptions struct {
+		Requests      *int    `yaml:"requests" json:"requests"`
+		Concurrency   *int    `yaml:"concurrency" json:"concurrency"`
+		GlobalTimeout *string `yaml:"globalTimeout" json:"globalTimeout"`
+	} `yaml:"runnerOptions" json:"runnerOptions"`
+}
```

Comment:
I see by the commit message that somehow using pointers to type fixes zero-value override.
As I did not see the bug happening, do you care to explain a bit ? :)

Reply:
Sure! In short, it's only a necessary step to make `parseRawConfig` where lies the fix:
https://github.com/benchttp/runner/blob/22cb9d071a839c652e6300b96bb8715a69516acd/config/file/parse.go#L35-L86

Just like for CLI flags with `main.configFlags`, we need to know which options are set regardless of their value. But this is much more complex to achieve for parsed json/yaml, so here we use pointers instead to determine whether a field is set. It's handy because their zero value is `nil`: without them it would be `""` or `0` and we wouldn't be able to tell if it's defaulted or an actual value from the config file.

Reply:
Yeah obviously ! relying on nil is a great way of doing this !